test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightDeserializerTests.cs

```diff
@@ -1144,25 +1145,385 @@ public void ParsingExpectedComplexPropertyActualNotShouldThrow()
+                (jsonReader, propertyName) => jsonReader.ReadPrimitiveValueAsync(),
+                async (jsonReader) =>
+                {
+                    await jsonReader.ReadAsync(); // Position the reader on the first node
+                    await jsonReader.ReadAsync(); // Read StartObject
+                },
+                (jsonReader, propertyParsingResult, propertyName) =>
+                {
+                    Assert.Equal("Colors", propertyName);
+                    Assert.Equal(
+                        ODataJsonLightDeserializer.PropertyParsingResult.PropertyWithoutValue,
+                        propertyParsingResult);
+
+                    return TaskUtils.CompletedTask;
+                });
+        }
+
+        [Theory]
+        [InlineData("13", true)]
+        [InlineData("\"13\"", false)]
+        public async Task ReadAndValidateAnnotationAsLongForIeee754CompatibleAsync_ThrowsExceptionForConflictBetweenInputFormatAndParameter(
+            string data,
+            bool isIeee754Compatible)
+        {
+            var payload = $"{{\"Data\":{data}}}";
+
+            using (var jsonInputContext = CreateJsonLightInputContext(payload, EdmCoreModel.Instance, isAsync: true, isIeee754Compatible: isIeee754Compatible))
+            {
+                var jsonLightDeserializer = new ODataJsonLightPropertyAndValueDeserializer(jsonInputContext);
+                await jsonLightDeserializer.JsonReader.ReadAsync(); // Position the reader on the first node
+                await jsonLightDeserializer.JsonReader.ReadAsync(); // Read StartObject
+                await jsonLightDeserializer.JsonReader.ReadPropertyNameAsync();
+
+                var exception = await Assert.ThrowsAsync<ODataException>(
+                    () => jsonLightDeserializer.ReadAndValidateAnnotationAsLongForIeee754CompatibleAsync("@odata.count"));
+
+                Assert.Equal(
+                    Strings.ODataJsonReaderUtils_ConflictBetweenInputFormatAndParameter("Edm.Int64"),
+                    exception.Message);
+            }
+        }
+
+        #endregion Async Tests
+
         private ODataJsonLightInputContext CreateJsonLightInputContext(string payload)
         {
             return this.CreateJsonLightInputContext(payload, this.edmModel);
         }
 
-        private ODataJsonLightInputContext CreateJsonLightInputContext(string payload, IEdmModel model, bool isResponse = true)
+        private ODataJsonLightInputContext CreateJsonLightInputContext(
+            string payload,
+            IEdmModel model,
+            bool isResponse = true,
+            bool isAsync = false,
+            bool isIeee754Compatible = false)
         {
             var messageInfo = new ODataMessageInfo
             {
                 IsResponse = isResponse,
-                MediaType = JsonLightUtils.JsonLightStreamingMediaType,
-                IsAsync = false,
+                MediaType = CreateMediaType(isIeee754Compatible),

```

Comment:
Why change the media type?

Reply:
For IEEE754Compatible `long` and `decimal` values to be returned by the service, the request header needs to include `IEEE754Compatible=true` in the `Content-Type` header; otherwise `IEEE754Compatible=false` (default) is included. By changing the media type, we're able to test both scenarios